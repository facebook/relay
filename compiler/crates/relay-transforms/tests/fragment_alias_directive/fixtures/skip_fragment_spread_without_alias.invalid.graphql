# expected-to-throw
fragment RelayReaderNamedFragmentsTest_user on User {
  name
}

query RelayReaderNamedFragmentsTest2Query($someCondition: Boolean!) {
  me {
    # This might not match!
    ...RelayReaderNamedFragmentsTest_user @skip(if: $someCondition)
  }
}