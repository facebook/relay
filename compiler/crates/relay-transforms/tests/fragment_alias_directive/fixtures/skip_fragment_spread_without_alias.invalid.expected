==================================== INPUT ====================================
# expected-to-throw
fragment RelayReaderNamedFragmentsTest_user on User {
  name
}

query RelayReaderNamedFragmentsTest2Query($someCondition: Boolean!) {
  me {
    # This might not match!
    ...RelayReaderNamedFragmentsTest_user @skip(if: $someCondition)
  }
}
==================================== ERROR ====================================
✖︎ Expected `@alias` directive. Fragment spreads with `@skip` are conditionally fetched. Add `@alias` to this spread to expose the fragment reference as a nullable property.

  skip_fragment_spread_without_alias.invalid.graphql:9:43
    8 │     # This might not match!
    9 │     ...RelayReaderNamedFragmentsTest_user @skip(if: $someCondition)
      │                                           ^^^^^
   10 │   }
